src/System.Buffers.Experimental/System/Buffers/Sequences/SequenceExtensions.cs

```diff
@@ -12,24 +12,38 @@ public static class Sequence
         public static ReadOnlySpan<byte> ToSpan(this ReadOnlySequence<byte> sequence)
         {
             SequencePosition position = sequence.Start;
-            ResizableArray<byte> array = new ResizableArray<byte>(1024);
+            int capacity = 0;
+            while (sequence.TryGet(ref position, out ReadOnlyMemory<byte> buffer))
```

Comment:
Is there any point (based on segment size and number of segments within the ROSequence) where iterating through it twice (in two loops) in your new solution is slower than the extra array allocations and data copies of using ResizableArray?

Maybe if we have 1_024 empty segments, the cost of the iterating twice would be higher than the allocation/data copy (since there would be nothing to copy). Might be worth measuring just to see the performance characteristics of edge cases.

Reply:
1024 segments gives the greatest disparity between the two methods.  Using ResizableArray is only slightly faster in that case.  For the smaller values, there is no discernible difference

Reply:
What happens if the segments are empty?

There are two factors, **segment size** and **number of segments**. I understand that 1024 number of segments give the greatest disparity and how lowering that doesn't impact the result, but what about changing the **segment size** all the way down to 0?